File: Source/Coco.Client/src/models/SignupModel.js
let model = {
  lastname: {
    value: "",
    validation: {
      isRequired: true
    },
    isValid: false
  },
  firstname: {
    value: "",
    validation: {
      isRequired: true
    },
    isValid: false
  },
  email: {
    value: "",
    validation: {
      isEmail: true
    },
    isValid: false
  },
  password: {
    value: "",
    validation: {
      isRequired: true
    },
    isValid: false
  },
  confirmPassword: {
    value: "",
    validation: {
      isRequired: true,
      sameRefProperty: "password"
    },
    isValid: false
  },
  genderId: {
    value: 1,
    validation: {
      isRequired: true
    },
    isValid: true
  },
  birthDate: {
    value: new Date(),
    validation: {
      isDate: true
    },
    isValid: false
  }
};

export default model;

File: src/Server/Modules/Identity/src/Module.Auth.Api/Models/SignupModel.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace  Module.Auth.Api.Models
{
    public class SignupModel
    {
        [Required]
        public string Lastname { get; set; }
        [Required]
        public string Firstname { get; set; }
        [Required]
        public string Email { get; set; }

        [Required]
        public string Password { get; set; }
        [Required]
        public string ConfirmPassword { get; set; }
        [Required]
        public DateTime? BirthDate { get; set; }
        public int? GenderId { get; set; }
    }
}
